### PR TITLE
CircleCI: fix op-node-docker-build

### PR DESCRIPTION
**Pull request by syntrust**
[Error log](https://app.circleci.com/pipelines/circleci/VJoJbnHRXJiFR13mWFvRfZ/SHyYDZiLFYMtveJSZVFKbw/65/workflows/59eb8321-99c7-42d9-92f2-d36029763d6b/jobs/1325): 

```log
ERROR: process "/bin/sh -c cd da-server && go build -o da-server main.go" did not complete successfully: exit code: 1
```

The error happens because there is no code to build, so we update the submodules before build.

### CODE REVIEW

**Comment by qzhodl** [comment]
I’m curious why the tests passed in the [PR](https://github.com/ethstorage/optimism/pull/124) without cloning the code. Could you clarify?

**Comment by syntrust** [comment]
> I’m curious why the tests passed in the [PR](https://github.com/ethstorage/optimism/pull/124) without cloning the code. Could you clarify?
 
From what I can figure out, the PR is tested using the Makefile but not with CircleCI.
The Makefile defines the dependency path as devnet-up: pre-devnet: submodules. Here`submodules` pulls code from the `da-server` using the command `git submodule update --init`.

However, in `.circleci/config.yml`, this command is only present in the `install-contracts-dependencies` step used by contract-related jobs.  
But `*-docker-build` jobs like `op-node-docker-build` in this issue build go code in Docker containers and have no submodule introduced before da-server.

**Comment by qzhodl** [comment]
> > I’m curious why the tests passed in the [PR](https://github.com/ethstorage/optimism/pull/124) without cloning the code. Could you clarify?
> 
> From what I can figure out, the PR is tested using the Makefile but not with CircleCI. The Makefile defines the dependency path as devnet-up: pre-devnet: submodules. Here`submodules` pulls code from the `da-server` using the command `git submodule update --init`.
> 
> However, in `.circleci/config.yml`, this command is only present in the `install-contracts-dependencies` step used by contract-related jobs. But `*-docker-build` jobs like `op-node-docker-build` in this issue build go code in Docker containers and have no submodule introduced before da-server.

Got it. I would suggest re-running the tests involved in the [PR](https://github.com/ethstorage/optimism/pull/124) to ensure that this change has no unintended side effects.

**Comment by syntrust** [comment]
> > > I’m curious why the tests passed in the [PR](https://github.com/ethstorage/optimism/pull/124) without cloning the code. Could you clarify?
> > 
> > 
> > From what I can figure out, the PR is tested using the Makefile but not with CircleCI. The Makefile defines the dependency path as devnet-up: pre-devnet: submodules. Here`submodules` pulls code from the `da-server` using the command `git submodule update --init`.
> > However, in `.circleci/config.yml`, this command is only present in the `install-contracts-dependencies` step used by contract-related jobs. But `*-docker-build` jobs like `op-node-docker-build` in this issue build go code in Docker containers and have no submodule introduced before da-server.
> 
> Got it. I would suggest re-running the tests involved in the [PR](https://github.com/ethstorage/optimism/pull/124) to ensure that this change has no unintended side effects.

In order to avoid the possible side effects and keep the impact in the CircleCI scope, I would suggest [this change](https://github.com/ethstorage/optimism/commit/958c58181bc4340a1dbbf8b101839300a73d03a8#diff-78a8a19706dbd2a4425dd72bdab0502ed7a2cef16365ab7030a5a0588927bf47R411) instead. It is also future-compatible for possible new submodules. Idea?

**Comment by qzhodl** [comment]
> > > > I’m curious why the tests passed in the [PR](https://github.com/ethstorage/optimism/pull/124) without cloning the code. Could you clarify?
> > > 
> > > 
> > > From what I can figure out, the PR is tested using the Makefile but not with CircleCI. The Makefile defines the dependency path as devnet-up: pre-devnet: submodules. Here`submodules` pulls code from the `da-server` using the command `git submodule update --init`.
> > > However, in `.circleci/config.yml`, this command is only present in the `install-contracts-dependencies` step used by contract-related jobs. But `*-docker-build` jobs like `op-node-docker-build` in this issue build go code in Docker containers and have no submodule introduced before da-server.
> > 
> > 
> > Got it. I would suggest re-running the tests involved in the [PR](https://github.com/ethstorage/optimism/pull/124) to ensure that this change has no unintended side effects.
> 
> In order to avoid the possible side effects and keep the impact in the CircleCI scope, I would suggest [this change](https://github.com/ethstorage/optimism/commit/958c58181bc4340a1dbbf8b101839300a73d03a8#diff-78a8a19706dbd2a4425dd72bdab0502ed7a2cef16365ab7030a5a0588927bf47R411) instead. It is also future-compatible for possible new submodules. Idea?

LGTM

File: .circleci/config.yml [hunk 1]
```diff
@@ -408,6 +408,8 @@ jobs:
       docker_layer_caching: true # we rely on this for faster builds, and actively warm it up for builds with common stages
     steps:
       - checkout
+      - run:
+          command: git submodule update --init
```
Comment (blockchaindevsh) [inline .circleci/config.yml:412]: Looks like `install-contracts-dependencies` also does this, can it be reused?

**UPDATE**

Reusing it is causing more lines of change, I'd prefer the previous one. Sorry for the back and forth.

Comment (syntrust) [reply .circleci/config.yml:412]: Yes, the previous is simple but the current is more structured, and they both pass the [test](https://app.circleci.com/pipelines/circleci/VJoJbnHRXJiFR13mWFvRfZ/SHyYDZiLFYMtveJSZVFKbw/75/workflows/8c45fb9a-d059-4580-94b5-b9869f515abe/jobs/1435). @qzhodl and @qizhou?

Comment (qzhodl) [reply .circleci/config.yml:412]: If I understand correctly, @blockchaindevsh was referring to not reusing the `git submodule update --init` command when mentioning “previous”

Comment (blockchaindevsh) [reply .circleci/config.yml:412]: Yeah, not reusing results in fewer lines of change here, will be easier to merge in future.

Comment (qzhodl) [reply .circleci/config.yml:412]: I agree, let's keep it simple

**Review by blockchaindevsh — APPROVED** [review]

**Review by qzhodl — APPROVED** [review]

**Pull request merged**